src/test/java/com/google/graphgeckos/dashboard/api/DashboardControllerTest.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(controllers = DashboardController.class)
+@AutoConfigureMockMvc
+public class DashboardControllerTest {
+
+  @MockBean
+  private DatastoreRepository datastoreRepository;
+
+  @Autowired
+  private MockMvc mvc;
+
+  private final String COMMIT_HASH = "1234";
+  private final Timestamp TIMESTAMP = Timestamp.now();
+  private final String BRANCH = "branch a";
+  private final GitHubData INITIAL_GITHUB_DATA = new GitHubData(COMMIT_HASH, TIMESTAMP, BRANCH);
+
+  /* Builders names. */
+  private final String NAME_A = "Builder A";
+  private final String NAME_B = "Builder B";
+
+  /* Builders statuses. */
+  private final String STATUS_A = "PASSED";
+  private final String STATUS_B = "FAILED";
+
+  /* Builders logs components. */
+  private final String LOG_TYPE_1 = "type 1";
+  private final String LOG_TYPE_2 = "type 2";
+  private final String LOG_TYPE_3 = "type 3";
+  private final String LOG_LINK_1 = "link 1";
+  private final String LOG_LINK_2 = "link 2";
+  private final String LOG_LINK_3 = "link 3";
+
+  /* Builders logs. */
+  private final List<Log> LOGS_A = Arrays.asList(new Log(LOG_TYPE_1, LOG_LINK_1),
+                                               new Log(LOG_TYPE_2, LOG_LINK_2));
+  private final List<Log> LOGS_B = Collections.singletonList(new Log(LOG_TYPE_3, LOG_LINK_3));
+
+  private final BuildBotData BUILDER_A = new BuildBotData(
+                                        COMMIT_HASH, NAME_A, LOGS_A, BuilderStatus.valueOf(STATUS_A));
+  private final BuildBotData BUILDER_B = new BuildBotData(
+                                         COMMIT_HASH, NAME_B, LOGS_B, BuilderStatus.valueOf(STATUS_B));
+
+  private final BuildInfo BUILD_INFO = new BuildInfo(INITIAL_GITHUB_DATA);
+
+
+  /* Illegal number of entries value. Revision must be >= 0. */
+  private final int MINUS_ONE_ENTRIES = -1;
+
+  /* Valid number of entries and offset values. */
+  private final int ZERO_ENTRIES = 0;
+  private final int ONE_ENTRIES = 1;
+  private final int OFFSET_ZERO = 0;
+
+  /** Enables Mockito.*/
+  @Before
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Before
+  public void setUpBuildInfo() {
+    BUILD_INFO.addBuilder(BUILDER_A);
+    BUILD_INFO.addBuilder(BUILDER_B);
+  }
+
+  /**
+   * Given: number of the entries to return = 0, any offset >= 0
+   * Expected behaviour: empty JSON list, Response Status 200 (OK).
+   */
+  @Test
+  public void whenGivenZeroRevisionsOffsetZeroReturnsEmptyJSONArray() throws Exception {
+
```

Comment:
nit: remove blank line at the start of the block

Reply:
It improves readability because names pf the tests are very long.